operators/tokenizer/bpe_utils.hpp

```diff
@@ -198,353 +299,435 @@ class TokenWithRegularExp {
-          m_text = m_text.substr(3);
+    // (?i:'s|'t|'re|'ve|'m|'ll|'d) (case insensitive)
+    if (categories.size() == 1 && categories[0] == "A"){
+      if ((m_text[0] == U'\'') && (m_text.size() > 1)) {
+        if ((m_text[1] == U's') || (m_text[1] == U't') ||
+            (m_text[1] == U'm') || (m_text[1] == U'd') ||
+            (m_text[1] == U'S') || (m_text[1] == U'T') ||
+            (m_text[1] == U'M') || (m_text[1] == U'D')) {
+          std::u32string res = ustring(m_text.substr(0, 2));
+          m_text = m_text.substr(2);
           return res;
         }
+
+        if (m_text.size() > 2) {
+          if ((((m_text[1] == U'r') || (m_text[1] == U'R')) && ((m_text[2] == U'e') || (m_text[2] == U'E'))) ||
+              (((m_text[1] == U'v') || (m_text[1] == U'V')) && ((m_text[2] == U'e') || (m_text[2] == U'E'))) ||
+              (((m_text[1] == U'l') || (m_text[1] == U'L')) && ((m_text[2] == U'l') || (m_text[2] == U'L')))) {
+            std::u32string res = ustring(m_text.substr(0, 3));
+            m_text = m_text.substr(3);
+            return res;
+          }
+        }
       }
     }
 
-    // Case 2: Custom Character Combination (2nd Alternative)
-    // \r matches a carriage return (ASCII 13)
-    // \n matches a line-feed (newline) character (ASCII 10)
-    // all other symbols as previously described.
-
-    // [^\r\n\p{L}\p{N}]?\p{L}+
-    if (!IsRN(m_text[0]) && !IsN(m_text[0])){
-      if (IsL(m_text[0]) || ((m_text.size() > 1) && IsL(m_text[1]))){
+    if (categories.size() == 1 && categories[0] == "sS") {
+      if ((m_text.size() >= 1) && (IsZ(m_text[0]))) {
         size_t i = 1;
         for (; i < m_text.size(); ++i) {
-          if (!IsL(m_text[i]))
-            break;
+          if (!IsZ(m_text[i])) break;
         }
-        std::u32string_view res = m_text.substr(0, i);
+        if ((i > 1) && (i != m_text.size())) {  //\s+(?!\S)
+          i--;
+          std::u32string res = ustring(m_text.substr(0, i));
+          m_text = m_text.substr(i);
+          return res;
+        }
+        // \s+
+        std::u32string res = ustring(m_text.substr(0, i));
         m_text = m_text.substr(i);
         return res;
       }
     }
 
-    // Case 3: Numeric Character Class (3rd Alternative)
-    // {1,3} matches the previous token between 1 and 3 times, as many times as possible, giving back as needed (greedy)
-    // all other symbols as previously described. 
+    return U"";
+  }
 
-    // \p{N}{1,3}
-    if (IsN(m_text[0])){
-      size_t i = 1;
-      for (; i < m_text.size(); ++i) {
-        if (!IsN(m_text[i]) || (i > 2))
-          break;
-      }
-      std::u32string_view res = m_text.substr(0, i);
-      m_text = m_text.substr(i);
-      return res;
-    }
+  // Function to wrap standalone \p{...} categories
+  std::string WrapStandaloneCategories(const std::string &input) {
+      std::string modified_input = input;
+      std::regex standalone_category_regex(R"(\\p\{[A-Za-z]+\})");
```

Comment:
it's already RawString ( R"(....)"), why is there still double \ ahead of p?

Reply:
i tried without double backslash, seems RawString does not interpret backslashes as escape characters